File: 9.Maximum Slice Problem - MaxSliceSum/answer.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    let currentMax = A[0]
    let highestMax = A[0]
    
    for (let i =0; i < A.length-1; i++){
        currentMax = Math.max(currentMax + A[i+1], A[i] + A[i+1])
        currentMax = Math.max(currentMax, A[i+1])
        highestMax = Math.max(currentMax, highestMax)
    }
    
    return highestMax
}

File: 12.Euclidean Algorithm - ChocolateByNumbers/answer.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")
def gcd(a, b):
    if a % b == 0:
        return b
    else:
        return gcd(b, a % b)
        
# Explanation: It has something to do with the math property, found it
# by trying out different values and found the pattern where the answer
# is always (higherValues / commonDivisor)
def solution(N, M):
    # write your code in Python 3.6
    commonDivisor = gcd(N, M)
    return N // commonDivisor
